types/stripe/index.d.ts

```diff
@@ -6816,6 +7223,140 @@ declare namespace Stripe {
+			): Promise<paymentIntents.IPaymentIntent>;
+			retrieve(
+				id: string,
+				response?: IResponseFn<paymentIntents.IPaymentIntent>,
+			): Promise<paymentIntents.IPaymentIntent>;
+
+			/**
+			 * Confirm that your customer intends to pay with current or provided `source`. Upon confirmation, the PaymentIntent will attempt to initiate a payment.
+			 *
+			 * If the selected source requires additional authentication steps, the PaymentIntent will transition to the `requires_action` status and suggest additional actions via `next_source_action`. If payment fails, the PaymentIntent will transition to the `requires_payment_method` status. If payment succeeds, the PaymentIntent will transition to the `succeeded` status (or `requires_capture`, if `capture_method` is set to `manual`).
+			 *
+			 * When using a publishable key, the client_secret must be provided to confirm the PaymentIntent.
+			 */
+			confirm(
+				paymentIntentId: string,
+				data: paymentIntents.IPaymentIntentConfirmOptions,
+				options: HeaderOptions,
+				response?: IResponseFn<paymentIntents.IPaymentIntent>,
+			): Promise<paymentIntents.IPaymentIntent>;
+			confirm(
+				paymentIntentId: string,
+				data: paymentIntents.IPaymentIntentConfirmOptions,
+				response?: IResponseFn<paymentIntents.IPaymentIntent>,
+			): Promise<paymentIntents.IPaymentIntent>;
+
+			/**
+			 * Capture the funds of an existing uncaptured PaymentIntent where `required_action="requires_capture"`.
+			 * Uncaptured PaymentIntents will be canceled exactly seven days after they are created.
+			 *
+			 * @returns Returns a PaymentIntent object with `status="succeeded"` if the PaymentIntent was capturable. Returns an error if the PaymentIntent was not capturable or an invalid amount to capture was provided.
+			 */
+			capture(
+				paymentIntentId: string,
+				data: paymentIntents.IPaymentIntentCaptureOptions,
+				options: HeaderOptions,
+				response?: IResponseFn<paymentIntents.IPaymentIntent>,
+			): Promise<paymentIntents.IPaymentIntent>;
+			capture(
+				paymentIntentId: string,
+				data: paymentIntents.IPaymentIntentCaptureOptions,
+				response?: IResponseFn<paymentIntents.IPaymentIntent>,
+			): Promise<paymentIntents.IPaymentIntent>;
+
+			/**
+			 * A PaymentIntent object can be canceled when it is in one of these statuses: `requires_payment_method`, `requires_capture`, `requires_confirmation`, `requires_action`.
+			 * Once canceled, no additional charges will be made by the PaymentIntent and any operations on the PaymentIntent will fail with an error. For PaymentIntents with `status='requires_capture'`, the remaining `amount_capturable` will automatically be refunded.
+			 *
+			 * @returns Returns a PaymentIntent object if the cancellation succeeded. Returns an error if the PaymentIntent has already been canceled or is not in a cancelable state.
+			 */
+			cancel(
+				paymentIntentId: string,
+				data: {
+					cancellation_reason?:
```

Comment:
I've seen references to some of those strings in the docs already, please create some new types and use them accordingly to avoid too much duplication.